developer_guides/firmware/async_messaging_best_practices.rst

```diff
@@ -0,0 +1,651 @@
+.. _async_messaging_best_practices:
+
+Async Messaging Best Practices
+##############################
+
+This section provides best practices and point out issues that developers should
+be aware of when using asynchronous messages in their components.
+
+Message Type IDs
+****************
+
+The only unique value that is guaranteed to not change is UUID. The message type
+IDs can change every time when a component is registering as a message
+consumer/producer. In fact it depends on initialization order of components and
+it may seem to be static. However it can change i.e. with addition of a new
+component to pipeline and the entire mechanism will stop working.
+
+The components must not hardcode this value and they must use a runtime value of
+component type ID returned during producer/consumer registration.
+
+.. _message_handling:
+
+Message Handling
+****************
+
+The asynchronous messages consumers must be prepared to handle the messages
+asynchronously. A message can be received while a component is processing data
+and it can lead to undefined behavior when internal state is changed
+immediately. The recommended way of handling asynchronous messages is to save a
+message and apply the changes during a next processing time i.e. at the
```

Comment:
who will save the message - should it be client consumer code ?

Reply:
> who will save the message - should it be client consumer code ?

The suggested solution is provided in a paragraph below to "implement a queue of incoming asynchronous messages in a
component".